### PR TITLE
modular.tsconfig

### PR DESCRIPTION
**Pull request by threepointone**
Fixes https://github.com/jpmorganchase/modular/issues/60

This creates a new package, `modular.tsconfig`, for usage with modular projects. We can iterate on this, and further improve it later. The benefit is that we can make meaningful changes and propagate them to consumers.

This PR does _not_ make changes to `create-modular-react-app` to use this just yet, because we'll have to publish this first before we can use it in tests, so I've punted on that for the next PR.

(PS: Much like jest/babel/etc configs, would've been nice if we could hide this and prevent changes altogether, but I don't think the community/ecosystem is mature enough for that yet. We'll tackle that at some point)

### CODE REVIEW

**Comment by LukeSheard** [comment]
> Much like jest/babel/etc configs, would've been nice if we could hide this and prevent changes altogether

Why not? I think this is a serious discussion we should have before merging this. Especially given that we're not beginning to start iterating on this. From the discussions we've had we know that once we make things too loose and "easy" there'll be no turning back. Maybe now this PR is where we start locking this stuff down so to speak.

**Comment by threepointone** [comment]
Yeah we can discuss this, it’s not super critical to land this just yet. Some quick notes: 
- firstly, we’ll have fairly strict defaults, not a problem. 
- but for organisations just starting the typescript journey, unless they have experts on their team, they would be wary of not having some flexibility while they become comfortable using it. Given an option of strict or nothing, they’ll choose nothing (ie- they won’t even use modular). Or they’ll just write .js files, bypassing typescript altogether.
- it’s another thing for teams that are already familiar with typescript; in such a case, it’s possible to put _another_ layer of strictness with other tooling (or simply, reject any PRs/changes to tsconfig.json).
- finally, I’m not sure it’s possible to configure a custom path for this configuration? Which means it’s not possible to hide it I think. Happy to be wrong, but the above points are still legit I think. 

Happy to discuss JP specific concerns in person too.

**Review by matt-d-webb — APPROVED** [review]

**Comment by threepointone** [comment]
Not going to merge this just yet, I'll do it after I release what we have in master, then work on integrating this into create-modular-react-app's install flow (and our own repo, ofc), and then release it. Tentatively this weekend, maybe early next week.

**Comment by threepointone** [comment]
Going to land this so I can get to work integrating it with the rest of the codebase/projects.

**Pull request merged**
